src/psyclone/domain/lfric/psyir.py

```diff
@@ -101,10 +101,10 @@
     # Create the specific symbol
     exec(
         "class {0}DataSymbol(DataSymbol):\n"
```

Comment:
This was modified because these dynamically created LFRic symbol classes didn't have 'visibility' as a constructor argument. This argument should have been here anyway but this became an issue now because the new_symbol method assigns an explicit visibility to new symbols depending on the symbol table default visibility.

Since visibility (and the interface argument which has here before) are not responsibilities of this class, I opted for just a kwargs argument which is passed to the super constructor.

Reply:
I added a visibility argument on some of the tests for this classes

Reply:
Aha, thanks for explaining.